marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -476,7 +478,12 @@ private RunStepResponse runIngester(RunStepResponse runStepResponse, Collection<
                 successfulEvents.addAndGet(batch.getItems().length-1);
                 successfulBatches.addAndGet(1);
                 logger.debug(String.format("Current SuccessfulEvents: %d - FailedEvents: %d", successfulEvents.get(), failedEvents.get()));
-                runStatusListener(successfulBatches.get()+failedBatches.get(), batchCount, successfulEvents, failedEvents);
+                if("csv".equalsIgnoreCase(inputFileType)) {
+                   runCsvStatusListener(uris.size(), successfulEvents, failedEvents);
+                }
+                else {
+                    runStatusListener(successfulBatches.get()+failedBatches.get(), batchCount, successfulEvents, failedEvents);
```

Comment:
A refactor to consider is that the successfulEvents, failedEvents, successfulBatches, and failedBatches variables are often passed to other methods. That's a good indication that they should be grouped together into a new object class - e.g. "StepStatus". 

That would also help avoid the duplication of this code block. You could instead have a method like this:

protected void runStatusListener(String inputFileType, Collection<String> uris, StepStatus stepStatus)

(I don't think batchCount needs to be passed in, as that can be quickly calculated whenever it's needed). 

That method would check to see if inputFileType == csv. You could then call that method here and in the block below so that the if/else isn't duplicated.